MNT: Compat with pytest>=8

pllim opened the pull request:
Because the hook check has changed in 8.1 . See https://github.com/pytest-dev/pytest/issues/11779#issuecomment-1879387034

bsipocz commented on pytest_doctestplus/plugin.py:
```diff
@@ -519,159 +519,299 @@ def parse(self, s, name=None):
+            Skip paths that match any of the doctest_norecursedirs patterns or
+            if doctest_only is True then skip all regular test files (eg test_*.py).
+            """
+            if PYTEST_GE_8_0:
+                dirpath = Path(path).parent
+                collect_ignore = config._getconftest_pathlist("collect_ignore",
+                                                              path=dirpath)
```
no need for this conditional

pllim replied:
Good catch! Accidentally left this in from an earlier attempt.

bsipocz commented on pytest_doctestplus/plugin.py:
```diff
@@ -519,159 +519,299 @@ def parse(self, s, name=None):
-                                                          path=dirpath)
-        elif PYTEST_GE_7_0:
-            dirpath = Path(path).parent
+if PYTEST_GE_8_0:
```
Could this conditional be in and around methods that's affected rather than for the whole class?

pllim replied:
I don't think so. If I understood https://github.com/pytest-dev/pytest/issues/11779#issuecomment-1879387034 correctly, I have to update the function signature, not just the internals.

bsipocz replied:
yeap, what I meant was to do the conditional outside of the affected methods, but not outside of the class.

pllim replied:
Oh, now that you mention it, I think Marten did something like that for astropy.units . Lemme try.

pllim replied:
Okay, that works though in the end, most of the class still need the version check. 🤷‍♀️

bsipocz replied:
ahh, I'm sorry about it then, hoped it would end up a bit cleaner.

bsipocz reviewed (commented):
I find having the conditional around the class makes things less readable, but if you say it's better this way, I'm OK with it.

That's being said, in the else branch there are some cleanups to be made for superfluous conditional.

bsipocz commented:
Oh, and please mention it in the changelog.

pllim commented:
@bsipocz , I think I addressed your comments. Also did some infrastructure updates that I noticed while porting stuff from here to https://github.com/astropy/pytest-filter-subpackage/pull/15 .

bsipocz commented on tox.ini:
```diff
@@ -28,9 +27,7 @@ deps =
     pytest73: pytest==7.3.*
     pytest74: pytest==7.4.*
```
I would already add a placeholder for the 8.0.x series (with a temporary `pre` option), given these changes affect that branch and currently, we don't have an idea whether all is fine, or it is only passing with 8.1.dev

bsipocz reviewed (approved):
The code changes look, good, but I would like to play it save and include 8.0.x in the test suite. Given it's already the evening for you, I'll just push a commit to this branch, and go ahead and merge if CI is all green.

bsipocz commented:
Thanks @pllim!

pllim commented:
Indeed I was AFK so thanks for wrapping this up! 🙇‍♀️
